application/backend/src/main/resources/apps/typerefinery/components/widgets/map/clientlibs/functions.js

```diff
@@ -60,13 +67,33 @@ window.Typerefinery.Page.Tms = Typerefinery.Page.Tms || {};
             ns.modelDataConnected($component);
         }
     }
+    //update map after tms connected
+    ns.updateMapInstance = (data, $component) => {
+        let componentConfig = componentNs.getComponentConfig($component);
+        const map = mapInstanceNs[componentConfig.id]
+        map.eachLayer((layer)=>{
+                if (layer instanceof leafletMap.Marker){
+                    map.removeLayer(layer)
+                }                
+        })
+        if(data.mapLat && data.mapLng && data.zoomLevel){
+            map.setView([data.mapLat, data.mapLng ], data.zoomLevel);
+        }
+        const markerDetails=data.markers
+        if(markerDetails.length>0){
+            ns.addMarker(data.markers, map)
```

Comment:
You are taking up data.markers and assigning it to markerDetails,
So why again using data.markers ?

Reply:
Replaced